### PR TITLE
nightly split releases

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
Because there's no reason not to. The only obstacle to automating the
normal release process is that we need an explicit manual validation
step for our audit log when creating a release, but split-releases are
created in the Assemblty repo, so we can have the audit log over there.

The diff is going to be very messy because there's a lot of stuff moving
around. The only thing that is not just moving a job to a separate file
is the `ci/daily-snapshot.yml` file, which is 100% new and is meant as
a new Azure Pipelines entrypoint (which I will create after this gets
approved). I have made a reasonable effort to create individual commits
that simplify reviewing, but I expect it's still going to be kind of a
mess. I'm open to opening separate PRs to ~bump my stats~ move one job
at a time if that makes reviewing (and testing) easier.

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: ci/daily-snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  steps:
+    - template: bash-lib.yml
+      parameters:
+        var_name: bash-lib
+    - bash: |
+        set -euo pipefail
+        eval "$(./dev-env/bin/dade-assist)"
+        source $(bash-lib)
+
+        release=$(./release.sh snapshot HEAD 0.0.0 | awk '{print $2}')
```
Comment (cocreature) [inline ci/daily-snapshot.yml:32]: I still feel slightly uneasy about this but don’t really have a better idea for now so let’s go with that.

**Review by cocreature — APPROVED** [review]
Looks sensible, thank you!

File: ci/daily-snapshot.yml (hunk 1 above)
Comment (garyverhaegen-da) [reply ci/daily-snapshot.yml:32]: The `0.0.0` prefix? Would you prefer something smarter like:
```
$(head -1 LATEST | awk '{print $2}' | sed -e 's/\([0-9]\+\.[0-9]\+\.[0-9]\+\).*/\1/')
```

Or just the general idea of having daily snapshots?

Comment (cocreature) [reply ci/daily-snapshot.yml:32]: Yeah it’s about the prefix. I kind of liked seeing very easily where a snapshot fits in before and while I can figure it out based on commit numbers & hashes, it’s a lot less convenient for humans.

**Review by stefanobaghino-da — COMMENTED** [review]
🎉

**Pull request merged**